cmd/peer-rest-server.go

```diff
@@ -575,182 +570,75 @@ func (s *peerRESTServer) DeleteBucketHandler(w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	globalNotificationSys.RemoveNotification(bucketName)
-	globalPolicySys.Remove(bucketName)
-	globalBucketObjectLockSys.Remove(bucketName)
-	globalBucketQuotaSys.Remove(bucketName)
-	globalLifecycleSys.Remove(bucketName)
-
+	globalBucketMetadataSys.Remove(bucketName)
 	w.(http.Flusher).Flush()
 }
 
-// ReloadFormatHandler - Reload Format.
-func (s *peerRESTServer) ReloadFormatHandler(w http.ResponseWriter, r *http.Request) {
+// LoadBucketMetadataHandler - reloads in memory bucket metadata
+func (s *peerRESTServer) LoadBucketMetadataHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
Instead of just sending the bucket name and having to fetch it from the object layer I would just send the serialized data I want it set to as the body.

If you need to be able to load, just make it optional if `len(body)==0`.

I will do final review once you have the last parts in.

Reply:
The reason I did it this way to always fetch the state from the disk, as that is the source of truth not the in-flight value over the network. Because the source of truth is very important in keeping this metadata sort of consistent across-the-board.

And only consistent entity is on the disk..

Reply:
Yeah, but doesn't the sender hold a lock while updating?

It just seems like a very roundabout way to sync up the cluster.

Reply:
It is not a global lock, we don't really lock things at handler it is only locked to avoid simultaneously updating the metadata file but there is room for in-transit calls 

The notification itself is not locked so it possible to be inconsistent. That's why sort of relying on what exists on disk.